src/preamble.js

```diff
@@ -1120,6 +1120,18 @@ function createWasm() {
       var exports = Module['instantiateWasm'](info, receiveInstance);
 #if ASYNCIFY
       exports = Asyncify.instrumentWasmExports(exports);
+#endif
+#if USE_OFFSET_CONVERTER
+      // We have no way to create an OffsetConverter in this code path since we
+      // have no access to the wasm binary (only the user does). Instead, create
+      // a fake one that reports we cannot identify functions from their binary
+      // offsets.
+      wasmOffsetConverter = {
+        getName: function() {
+          return 'unknown-due-to-instantiateWasm';
+        }
+      };
+      {{{ runOnMainThread("removeRunDependency('offset-converter');") }}}
```

Comment:
Does this effect threaded programs? IIRC we use instantiateWasm in worker.js to inject the module on pthreads.

Reply:
It doesn't affect them, as we send workers the offset data. But I refactored this to do the new work only on the main thread, which is clearer I think.